python/cuml/linear_model/linear_regression.pyx

```diff
@@ -65,6 +67,48 @@ cdef extern from "cuml/linear_model/glm.hpp" namespace "ML::GLM":
                      double *sample_weight) except +
 
 
+def divide_non_zero(x1, x2):
+    # Value chosen to be consistent with the RAFT implementation in
+    # linalg/detail/lstsq.cuh
+    eps = 1e-10
+
+    # Do not divide by values of x2 that are smaller than eps
+    mask = abs(x2) < eps
+    x2[mask] = 1.
+
+    return x1 / x2
+
+
+def fit_multi_target(X, y, fit_intercept=True, sample_weight=None):
+    assert X.ndim == 2
+    assert y.ndim == 2
+
+    assert X.shape[1] > 0, "Number of columns cannot be less than one"
+    assert X.shape[0] > 1, "Number of rows cannot be less than two"
```

Comment:
To validate user input it is better to test it explicitly and raise an exception (assertion are turned off if one enables optimization).